internal/controller/pkg/resolver/reconciler.go

```diff
@@ -64,6 +64,7 @@ const (
 	errSortDAG              = "cannot sort DAG"
 	errFmtMissingDependency = "missing package (%s) is not a dependency"
 	errInvalidConstraint    = "version constraint on dependency is invalid"
+	errDowngradeNotAllowed  = "downgrading package version is not allowed"
```

Comment:
I suggest rephrasing this to capture the nuance a little better.

I think ideally the full error string would communicate:

* The currently installed version is incompatible with a new package Crossplane is trying to install
* To make the versions compatible, Crossplane needs to downgrade this package
* Crossplane is refusing to downgrade this package automatically
* You're allowed to manually downgrade the package in order to unblock the new dependency

Reply:
I updated the error message and added event recording on lock object to make it visible to the user